enforcer/acls/acls.go

```diff
@@ -149,11 +149,31 @@ func (c *ACLCache) GetMatchingAction(ip []byte, port uint16) (*policy.FlowPolicy
 			// Scan the ports - TODO: better algorithm needed here
 			for _, p := range actionList {
 				if port >= p.min && port <= p.max {
-					return p.policy, nil
+
+					// Check observed policies.
+					if report == nil {
+						report = p.policy
+						packet = report
+						if p.policy.ObserveAction.ObserveContinue() {
+							continue
+						} else if p.policy.ObserveAction.ObserveApply() {
+							return report, packet, nil
+						}
+					}
```

Comment:
I think you will miss observed policies here. They are not sorted. I think its better if we create two different tables. One for observed policies and one for actionable policies and parse in parallel, even in a go-routing.